finality-provider/service/fp_instance.go

```diff
@@ -725,7 +725,13 @@ func (fp *FinalityProviderInstance) SubmitFinalitySignature(b *types.BlockInfo)
 	// get inclusion proof
 	proofBytes, err := fp.pubRandState.GetPubRandProof(pubRand)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get inclusion proof of public randomness: %v", err)
+		return nil, fmt.Errorf(
+			"failed to get inclusion proof of public randomness %s for FP %s for block %d: %v",
```

Comment:
```suggestion
			"failed to get inclusion proof of public randomness %s for FP %s for block %d: %w",
```
nit: error can better be wrapped